olp-cpp-sdk-core/src/cache/DiskCache.cpp

```diff
@@ -445,5 +409,32 @@ bool DiskCache::RemoveKeysWithPrefix(const std::string& prefix,
   return result.IsSuccessful();
 }
 
+void DiskCache::InitializeDB(const StorageSettings& settings,
+                             const std::string& path) const {
+  std::unique_ptr<leveldb::DB> database;
+  leveldb::DB* db = nullptr;
+
+  auto open_options = CreateOpenOptions(settings, false);
+  leveldb::DB::Open(open_options, path, &db);
+  database.reset(db);
+}
+
+leveldb::Options DiskCache::CreateOpenOptions(const StorageSettings& settings,
+                                              bool is_read_only) const {
+  leveldb::Options options;
+  options.compression = settings.compression;
+  options.info_log = leveldb_logger_.get();
+  options.write_buffer_size = settings.max_chunk_size;
+  options.filter_policy = leveldb::NewBloomFilterPolicy(10);
+  options.create_if_missing = !is_read_only;
```

Comment:
We should also create the db if it is not there.

Reply:
We are using `ReadOnlyEnv` for protected cache. It can't create files.

So only if cache folder was created, we also create db manually with `InitializeDB()`.